### ISSUE TITLE
[Question] How can I display an image for each device

### ISSUE THREAD

**Issue by snir99** [thingsboard/thingsboard#4362]
**Component**
 * UI

**Description**
I have a dashboard with many devices, each device has it unique image. What I want to do is display the image for each device but I want the implementation to be generic. Is there a way to send and image to the platform and display it?
I know I can put a single image in each dashboard state but because I have many devices with unique image for each one this will require many dashboard states, furthermore navigation through tables would not be possible because each device will have it's own dashboard state instead of one that will represent a group of them.
What can I do?

**Environment**
 * OS:  Ubuntu 20.04
 * ThingsBoard 3.2 CE

**Comment by kalutkaz** [comment]
Hi,
If you want to display some image for each device, you can save links of your image in device attributes.
![image](https://user-images.githubusercontent.com/43555187/113549285-cb970700-95f9-11eb-9969-7083a43676ec.png)
then add this attribute to your widget
![image](https://user-images.githubusercontent.com/43555187/113549528-45c78b80-95fa-11eb-971a-3539fdcfc114.png)
and set the attribute in src in the img tag
![image](https://user-images.githubusercontent.com/43555187/113550132-3c8aee80-95fb-11eb-833e-c002505752c0.png)
as you can see, widget display new image
![image](https://user-images.githubusercontent.com/43555187/113552111-5c6fe180-95fe-11eb-9a4f-0dbb48918d06.png)
also, you can use "Update server image attribute" widget. Add an attribute for img to this widget
![image](https://user-images.githubusercontent.com/43555187/113553954-339d1b80-9601-11eb-8b88-6554a573575e.png)
and you will be able to select an image from your computer to your device (image save in base64 format)
![image](https://user-images.githubusercontent.com/43555187/113552810-5af2e900-95ff-11eb-8a82-14461bf45c56.png)

**Comment by snir99** [comment]
@kalutkaz thanks for the detailed answer and sorry for the late reply I just got back to that issue.
I tried doing what you suggested, use "Update server image attribute" to Add an attribute for imageLink, but the image does not show up. I do see that the image is saved in the server attributes in base64 as you said but I cannot see it.
This is how I see it in the dashboard:
![dashboard-image-view](https://user-images.githubusercontent.com/31441724/122655756-a1b3a200-d15d-11eb-947b-bceac42ef80d.png)

This is how the image is saved in the server attributes:
![server-attributes-image-link](https://user-images.githubusercontent.com/31441724/122655767-ba23bc80-d15d-11eb-9143-80cc9ada0810.png)

This is the html card and the image tag I use there to try to display the image:
![dashboard-image-html-card](https://user-images.githubusercontent.com/31441724/122655780-d0ca1380-d15d-11eb-8a24-c5df8a420e2a.png)

What did I do wrong, do I need to use a different widget to set the image and not the static html card?
